### PR TITLE
Feature/xm table

### PR DESCRIPTION
**Pull request by EKrasovskiy**
Dynamic table

### CODE REVIEW

File: packages/components/table/xm-table/components/selection-column/selection-column.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public isAllSelected(): boolean {
+        const numSelected = this.selection.selected.length;
+        const numRows = this.dataSource?.data?.length;
+        return numSelected == numRows;
+    }
+
+    /** Selects all rows if they are not all selected; otherwise clear selection. */
+    public toggleAllRows(): void {
+
+        // this._selection.toggleAll(this.isAllSelected());
```
Comment (Kaskyi) [inline packages/components/table/xm-table/components/selection-column/selection-column.component.ts:43]: Make request to get all entites with pagination and mark them ass selected

File: packages/components/table/xm-table/components/table-header/table-actions-buttons/table-actions-buttons.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { XmTableSelectionService } from '@xm-ngx/components/table/xm-table/service/xm-table-selection-service/xm-table-selection.service';
+
+@Component({
+    selector: 'xm-table-actions-buttons',
+    templateUrl: './table-actions-buttons.component.html',
+    styleUrls: ['./table-actions-buttons.component.scss'],
+})
+export class TableActionsButtonsComponent implements OnInit {
+    @Input() public config: any;
```
Comment (Kaskyi) [inline packages/components/table/xm-table/components/table-header/table-actions-buttons/table-actions-buttons.component.ts:10]: TODO

File: packages/components/table/xm-table/components/table-header/table-filter-chips/table-filter-chips.component.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'xm-table-filter-chips',
+    templateUrl: './table-filter-chips.component.html',
+    styleUrls: ['./table-filter-chips.component.scss'],
+})
+export class TableFilterChipsComponent implements OnInit {
+
+    constructor() {
```
Comment (Kaskyi) [inline packages/components/table/xm-table/components/table-header/table-filter-chips/table-filter-chips.component.ts:10]: TODO:

File: packages/components/table/xm-table/components/table-header/table-filter/filter-dialog/filter-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<dav mat-dialog-title>Some CHIPS</dav>
+<mat-dialog-content>
+  <xm-form-layout [options]="config"
+                  [value]="group"></xm-form-layout>
+
+</mat-dialog-content>
+<mat-dialog-actions class="actionButtons">
+  <a>LEARN MORE</a>
```
Comment (Kaskyi) [inline packages/components/table/xm-table/components/table-header/table-filter/filter-dialog/filter-dialog.component.html:8]: TODO:i18n

**Comment by Kaskyi** [comment]
theme switch

File: packages/components/button-spinner/button-spinner.directive.ts [hunk 5]
```diff
@@ -32,12 +33,14 @@ export class ButtonSpinnerDirective implements OnChanges {
...
     constructor(
-        private matButton: MatButton,
+       @Optional() private matButton: MatButton,
+       @Optional() private matIconButton: MatIconButton,
         private componentFactoryResolver: ComponentFactoryResolver,
         private viewContainerRef: ViewContainerRef,
         private renderer: Renderer2,
     ) {
         this.spinnerFactory = this.componentFactoryResolver.resolveComponentFactory(MatProgressSpinner);
+        this.matButton = this.matButton || this.matIconButton;
```
Comment (Kaskyi) [inline packages/components/button-spinner/button-spinner.directive.ts:43]: merge into main with a separate commit

File: packages/components/entity-collection/http-client.rest.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ import { QueryParams } from './i-entity-collection';
 export class HttpClientRest<T extends IId = unknown, Extra extends Pageable = Pageable> implements IEntityCollectionPageable<T, Extra> {
 
     public readonly loading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
-    public readonly url: string;
+    public url: string;
```
Comment (Kaskyi) [inline packages/components/entity-collection/http-client.rest.ts:11]: @Injectable()
export class XmTableEntityRepository<T extends XmEntity>
    extends HttpClientRest<T, PageableAndSortable> {
    constructor(httpClient: HttpClient, config: XmTableConfigController) {
        super(null, httpClient);
    this.resourceUrl = this.config.collectton.reposito.Url;
    }

    private set resourceUrl(value: string) {
        this.url = value;
    }

File: packages/components/table/table/components/chips-control/chips-control.component.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+export const ChipsControlConfigDefault: ChipsControlConfig = {
+    elasticType: 'chips'
+};
+
+@Component({
+    standalone: true,
+    selector: 'xm-chips-control',
+    template: `
+        <!--        TODO: should remove after review-->
```
Comment (Kaskyi) [inline packages/components/table/table/components/chips-control/chips-control.component.ts:37]: ODO: should remove after review

File: packages/components/table/table/components/xm-table-empty.component.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+import { EmptyTableConfig } from '../interfaces/xm-table.model';
+import { NoDataModule } from '@xm-ngx/components/no-data';
+
+export interface TableNoDataRawManager<T extends CdkNoDataRow = CdkNoDataRow> {
+    setNoDataRow(columnDef: T): void;
+}
+
+
+export interface XmTableEmptyRows {
+    initial: EmptyTableConfig, // case , when table initially loaded empty
```
Comment (Kaskyi) [inline packages/components/table/table/components/xm-table-empty.component.ts:16]: tsdoc

Comment (Kaskyi) [reply packages/components/table/table/components/xm-table-empty.component.ts:16]: /** case , when table initially loaded empty **/

File: packages/components/table/table/interfaces/xm-table.model.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    //Table title
+    // options: XmTableOptions, // Table configuration
+    // dataSource: any,//XmTableDataSource, //Table data source specification
+}
+
+
+export interface XmTableDataSource {
+    /* data source type
```
Comment (Kaskyi) [inline packages/components/table/table/interfaces/xm-table.model.ts:58]: implement or rewrite. documentattiom docs

File: packages/shared/operators/src/format-with-config.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { transformByMap } from './transform-by-map';
+import { format } from './format';
+
+export interface FormatWithConfigOptions {
+    format?: unknown,
+    format2?: unknown
+}
+
+export function formatWithConfig<T>(data: unknown, config: FormatWithConfigOptions): T {
```
Comment (Kaskyi) [inline packages/shared/operators/src/format-with-config.ts:9]: delete

Comment (Kaskyi) [reply packages/shared/operators/src/format-with-config.ts:9]: " {{ \". " js

File: packages/components/xm-table.registry.ts [hunk 11]
```diff
@@ -1,8 +1,25 @@
...
+    {
+        selector: '@xm-ngx/components/table',
+        loadChildren: () => import('@xm-ngx/components/table/table/xm-table.component').then(m => m.XmTableComponent)
+    },
+    {
+        selector: '@xm-ngx/components/table-edit-cell',
+        loadChildren: () => import('@xm-ngx/components/table/components/xm-table-edit-cell.component').then(m => m.XmTableEditCellComponent)
+    },
+    {
+        selector: '@xm-ngx/components/edit-chips',
```
Comment (Kaskyi) [inline packages/components/xm-table.registry.ts:18]: should we move chips in a separate folder ?

**Review by oscaryald — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

**Review by Bozonych — APPROVED** [review]

**Pull request merged**